coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,272 @@
+# Coding Guidelines
+## Table of Contents
+
+- [General](#general)
+- [Comments](#comments)
+- [Strings](#strings)
+- [Variables and Attribute names](#variables-and-attribute-names)
+- [Object Literals](#object-literals)
+- [Indentation and Line-Breaks](#indentation-and-line-breaks)
+- [Functions](#functions)
+- [Control Statements](#control-statements)
+- [JSDoc](#jsdoc)
+- [JSX](#jsx)
+	- [Self contained elements](#self-contained-elements)
+	- [Non self contained elements](#non-self-contained-elements)
+
+## General
+We are using the eslintrc to lint our code. All basic rules are defined there, some important rules are:
+  - Max line length is 80 characters.
+  - Indentation is 2 spaces.
+  - No multiple empty lines.
+
+We prefer a functional approach and try to use the abstractions over
+normal control statements. Functions and code must be
+self-descriptive, use JSDoc blocks only if necessary.
+
+## Comments
+Rules:
+- Comments must be prefixed with a space sign.
+- The `*` sign is a exception to the rule.
+
+Examples:
+```JS
+// Single line Comment
+
+//* A Exception to the space rule
+
+/**
+* Block Comment
+*/
+```
+
+## Strings
+ - Strings are defined with double quotes.
+ - Use template strings if variables are used.
+
+Example:
+```JS
+var string = `Hello ${world}!`;
+```
+
+## Variables and Attribute names
+Variable names always use camelCase names, except Constants which use UPPER_CASE.
+One var, let or const per variable and line.
+
+Examples:
+```JS
+var camelCase = "{value}";
+let CamelCase = "{value}";
```

Comment:
For total correctness let's change this example to 
```let CamelCase = new Object();```
Var's that are starting with an upper case letter are intended to be created with the ```new``` keyword or are already referenced to the ```this``` keyword at an early point.